wtransport/src/config.rs

```diff
@@ -947,7 +957,10 @@ impl ClientConfigBuilder<states::WantsTransportConfigClient> {
     /// Completes configuration process.
     #[must_use]
     pub fn build(self) -> ClientConfig {
-        let mut quic_config = QuicClientConfig::new(Arc::new(self.0.tls_config));
+        let mut quic_config = QuicClientConfig::new(Arc::new(
+            quinn::crypto::rustls::QuicClientConfig::try_from(self.0.tls_config)
+                .expect("TLS 1.3 cipher suite missing"),
```

Comment:
https://docs.rs/quinn/latest/quinn/crypto/rustls/struct.QuicClientConfig.html#impl-TryFrom%3CClientConfig%3E-for-QuicClientConfig

It seems that:
> The initial cipher suite (AES-128-GCM-SHA256) is not available

If I am not wrong, it means we panic if `self.0.tls_config` does not contains `TLS13_AES_128_GCM_SHA256` cipher?

Reply:
I believe you are correct but, informally, this the same as if TLS 1.3 isn't supported (as I documented).
> /// Allows for manual configuration of a custom TLS setup using a provided
> /// [`rustls::ServerConfig`], **which must support TLS 1.3.**

Reply:
Maybe I am missing something, why is it the same?

Reply:
I mean users don't typically need to write `TLS13_AES_128_GCM_SHA256` in the program. Informally, [if they set TLS versions](https://docs.rs/quinn-proto/0.11.6/src/quinn_proto/crypto/rustls.rs.html#452), those version must include `rustls::version::TLS13`. [The default versions do include it](https://github.com/rustls/rustls/blob/f17866cd4040d7e38e3f1d84b16c671ef78b5ba7/rustls/src/versions.rs#L34-L46).

Reply:
However this may only hold for the `ring` provider. I suppose other providers could use a different TLS 1.3 cipher suite than `quinn` expects. Do you want me to try to explain this in the `wtransport` docs?